community/io/src/main/java/org/neo4j/io/pagecache/impl/muninn/MuninnReadPageCursor.java

```diff
@@ -56,9 +56,9 @@ public boolean next() throws IOException
         {
             return false;
         }
-        pin( nextPageId, false );
         currentPageId = nextPageId;
         nextPageId++;
+        pin( currentPageId, false );
```

Comment:
why do we move page pin below assigning current page id?

Reply:
Because `pin` will set `currentPageId = UNMAPPED_PAGE_ID` in `uncommonPin` when `PF_NO_FAULT` is enabled.